### DISCUSSION TITLE
what's the order of data when I fetch data using JavaScript?

### DISCUSSION THREAD

**Discussion by ascendho** [supabase/supabase#12536] [Questions] [answered]
## It is first time for me to use supabase, so I have one table here:
![image](https://user-images.githubusercontent.com/78126591/219945474-00abfb46-f31d-49f4-8b25-1af6a155c177.png)
## and I try to get these data using `fetch` method in JavaScript, actually I made it, but the order of data confused me, 
![image](https://user-images.githubusercontent.com/78126591/219945879-5c76ea00-b716-474d-9e04-0fa2ca35daaa.png)

**Comment by GaryAustin1** [comment, marked as answer]
Postgres (the database used by Supabase) has no natural order to the rows.  You could fetch them in a loop and get different order back (at least to some degree) each time.
If you want order, you have to have a column with a unique records that can be sorted.
You then use the .order() filter from the API to sort them on that column.

**Reply by ascendho** [reply]
Thank you!But could you give me a reference link?

**Reply by GaryAustin1** [reply]
It is just Postgres... I googled `postgres order`
And get https://www.postgresql.org/docs/current/queries-order.html

`If sorting is not chosen, the rows will be returned in an unspecified order. The actual order in that case will depend on the scan and join plan types and the order on disk, but it must not be relied on. A particular output ordering can only be guaranteed if the sort step is explicitly chosen.`

**Reply by ascendho** [reply]
No, I mean I want to know how to specify the order of data in JavaScript.

**Reply by ascendho** [reply]
You said the order() function.

**Reply by GaryAustin1** [reply]
I typed `order` in supabase.com 's documents page and get this:
https://supabase.com/docs/reference/javascript/order

**Reply by ascendho** [reply]
Ok, thanks

**Reply by GaryAustin1** [reply]
Just curious, you unmarked as the answer.  Do you need more info?

**Reply by ascendho** [reply]
Maybe I just forgot it and went to sleep.
